Sources/DeepLink/WebPurchaseRedemptionResult.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  WebPurchaseRedemptionResult.swift
+//
+//  Created by Antonio Rico Diez on 29/10/24.
+
+import Foundation
+
+/// Represents the result of a web purchase redemption
+/// - Seealso: ``Purchases/redeemWebPurchase(_:)``
+@objc public class WebPurchaseRedemptionResult: NSObject {
```

Comment:
Similar structure to the one used for `DeepLink`. I wanted to make this compatible with ObjC, and still seal the possible values so this was the best thing I found since I think we can't use enums with properties and keep it compatible with ObjC?

Reply:
The typical way of doing this in Objc is to pass a separate success and error completion blocks, but what we've been doing instead is passing a single block that has multiple parameters, one of them being an error e.g. `@escaping (CustomerInfo?, PublicError?)`. I think that could work in this case.

Reply:
Main concern with that approach is that we want to add more error cases in the future, for example, if the token is expired or already redeemed. I was thinking of having those be a different subclass of the result... otherwise, they would need to be part of the error and we would need to return a special type of error.

After talking with Andy, what I was thinking was to create 2 APIs, one swift-only that uses an enum and is nicer and one for ObjC which could look similar to the one currently in this PR? Wdyt?